### PR TITLE
[bad merge] py-pyprecice: Add version 2.1.1.2 and 2.2.0.1 

### PR DESCRIPTION
**Pull request by ajaust**
This adds the updated Python bindings for the current versions preCICE (https://github.com/spack/spack/pull/19264 and https://github.com/spack/spack/pull/21166)

The installation procedure has been improved such that we need the patch `deactivate-version-check-via-pip.patch` only for versions `<= 2.1.1.1` of the bindings. This is reflected accordingly in the Spack recipe.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 1]
```diff
@@ -44,7 +51,7 @@ class PyPyprecice(PythonPackage):
     depends_on("py-mpi4py", type=("build", "run"), when="+mpi")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py]: I won't think the `wheel` dep is needed

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py]: Also, the `py-mpi4py` package is always required, so I would remove the variant.

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 2]
```diff
@@ -44,7 +51,7 @@ class PyPyprecice(PythonPackage):
...
     depends_on("py-cython@0.29:", type=("build"))
 
-    phases = ['build_ext', 'install']
+    phases = ['install_lib', 'build_ext', 'install']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py]: Why is this change needed?

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 3]
```diff
@@ -53,4 +60,5 @@ def build_ext_args(self, spec, prefix):
         ]
 
     def install(self, spec, prefix):
-        self.setup_py("install", "--prefix={0}".format(prefix))
+        if self.version <= Version("2.1.1.1"):
+            self.setup_py("install", "--prefix={0}".format(prefix))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py]: Does the default install method not work for some reason?

File: var/spack/repos/builtin/packages/py-pyprecice/package.py (hunk 2 above)
Comment (BenjaminRodenberg) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py]: We switched from a single python module (only [`precice` in v2.1.1.2 and older](https://github.com/precice/python-bindings/blob/b5a2c82d4577cd637f7fdf63f68e3a4ccb1746a2/setup.py#L137)) to having two modules:

* [`precice`](https://github.com/precice/python-bindings/blob/d5383d6e4be2424aacbb0d24ab1e9c2a8212713b/setup.py#L151) is the actual user interface
* [`cyprecice`](https://github.com/precice/python-bindings/blob/d5383d6e4be2424aacbb0d24ab1e9c2a8212713b/setup.py#L79-L86) provides the actual implementation

This was necessary, since we started using `versioneer` to maintain versions (https://github.com/precice/python-bindings/pull/70) and this could not be done directly in the cython extension.

I tried building the spack module without the `install_lib` phase, but this was not successful. Then I [added this stage](https://github.com/precice/python-bindings/pull/68/commits/cc893289bb97c426c5c48a74fdb2fa6fda5cbdff) and it worked.

File: var/spack/repos/builtin/packages/py-pyprecice/package.py (hunk 3 above)
Comment (BenjaminRodenberg) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py]: We had to use the non-default method due to https://github.com/precice/python-bindings/issues/64 in older versions. In Version `2.2.0.1` this issue is fixed, but we still have to use the non-default method for compatibility reasons. Or is there a better way how to do this in spack?

Here is the discussion from our original PR: https://github.com/spack/spack/pull/19558#discussion_r513123239.

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 4]
```diff
@@ -53,4 +60,5 @@ def build_ext_args(self, spec, prefix):
         ]
 
     def install(self, spec, prefix):
-        self.setup_py("install", "--prefix={0}".format(prefix))
+        if self.version <= Version("2.1.1.1"):
```
Comment (BenjaminRodenberg) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py:66]:
```suggestion
        if self.version <= Version("2.1.1.1"):  # Needed for compatibility reasons. See https://github.com/spack/spack/pull/21881
```

Comment (BenjaminRodenberg) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py:66]: Maybe a comment here will help us in the future?

**Comment by adamjstewart** [comment]
Looks like a bad merge. Might want to make a new PR.

**Comment by ajaust** [comment]
Yeah, seems like I did something wrong. Sorry for the noise. We will fix the recipe and reopen the PR.
